src/app/routes/applications/new-role/new-role.component.ts

```diff
@@ -210,8 +210,8 @@ export class NewRoleComponent implements OnInit, AfterViewInit, OnDestroy {
       let parentNamespace = arrParentNamespace[1].substring(1);
 
       // Construct Version
-      let arrVersion = def.version.split('.');
-      let version = `${parseInt(arrVersion[0]) + 1}.0.0`;
+      const arrVersion = typeof(def.version) === 'string' ? parseInt(def.version.split('.')[0], 10) : def.version;
```

Comment:
Could you extract this to a method? So those two lines disappear from here and in patching form you can assign result of this method to `version`.

Reply:
@DawidGil-jlabs thanks, good tip. I updated the PR. Can you take a look and see if it is what you had in mind when you have time?

Reply:
Yes, this is what i had in mind. Thanks!